### ISSUE TITLE
VaProfile caching strategy discovery

### ISSUE THREAD

**Issue by LindseySaari** [department-of-veterans-affairs/va.gov-team#64549]
## Description

The current eMIS classes cache some of the data returned. For our migration to VaProfile, we will want to do some discovery around if we would benefit from caching. While I don't think prefill will benefit from caching, the other calls to VaProfile outside of prefill may benefit from a caching strategy.

Items to look at:
1. Look at coverband - the veteran_status info (veteran?, military_person?,  served_in_military?) - See how often these lines are called
2. Review the va_profile_redis models. Adjust our current (non-cached classes if needed)

**Comment by LindseySaari** [comment]
If we do see a benefit from caching, there's a current VaProfile redis caching class that we can reference. This class essentially calls the service and then caches the data [via this class, which invokes the service class](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/va_profile_redis/contact_information.rb#L190)

**Comment by laineymajor** [comment]
Work has not been started. Moving to next sprint.

**Comment by LindseySaari** [comment]
For the caching, it looks like there's a [good example in this class](https://github.com/department-of-veterans-affairs/vets-api/blob/v2_va_profile_veteran_status/app/models/va_profile_redis/contact_information.rb#L19) and how they used caching for VaProfile info

Essentially, we want to see if we will benefit from caching. Anywhere in the code that calls  `user.veteran_status` , `user.military_person?` or `user.veteran?` makes a call to VaProfile to pull VeteranStatus info. If those methods are called frequently for a user, it may be worthwhile to cache.

[This is the class](https://github.com/department-of-veterans-affairs/vets-api/blob/v2_va_profile_veteran_status/lib/va_profile/veteran_status/service.rb) I think could possibly benefit from caching

**Comment by LindseySaari** [comment]
Rachal is going to [run this code](https://github.com/department-of-veterans-affairs/vets-api/blob/API-27882_replace_evss_docs/rakelib/redis.rake#L83-L102) in prod to see how many existing cache entries we have for VeteranStatus info to benchmark if we will benefit from a caching strategy in the new VaProfile implementation

**Comment by RachalCassity** [comment]
```
 Audit Veteran Status results:

Total cached EMIS responses: 0
Veterans: 0
```

Something seems wrong here

**Comment by LindseySaari** [comment]
Rachal is continuing to work through the caching discovery and implementation (if deemed necessary)

**Comment by RachalCassity** [comment]
Still returning 0 or veteran status emis reponses

**Comment by rmtolmach** [comment]
Thanks, Rachal! @RachalCassity will close this tomorrow.

**Comment by LindseySaari** [comment]
Close this ticket and create a new caching ticket @LindseySaari

**Comment by laineymajor** [comment]
@RachalCassity please link the other ticket, then this can be closed.
Will proceed with the caching work once Ryan's ticket is complete.

**Comment by laineymajor** [comment]
This will be started once veteran status is complete

**Comment by RachalCassity** [comment]
@LindseySaari @laineymajor https://app.zenhub.com/workspaces/platform-product-team-633af4074573d06c3cda142a/issues/gh/department-of-veterans-affairs/va.gov-team/67106

**Comment by RachalCassity** [comment]
No longer discovery, new ticket linked above @laineymajor
